### ISSUE TITLE
Handle missing constructor more cleanly

### ISSUE THREAD

**Issue by matanlurey** [dart-lang/json_serializable#7]
_From @kevmoo on March 3, 2016 22:2_

Input

``` dart

@JsonSerializable()
class TestDoneEvent extends Object
    with _$TestDoneEventSerializerMixin  {
  @override
  final int time;

  @override
  String get type => "testDone";

  // The ID of the test that completed.
  final int testID;

  // The result of the test.
  final String result;

  // Whether the test's result should be hidden.
  final bool hidden;


  factory TestDoneEvent.fromJson(Map<String, dynamic> json) =>
      _$TestDoneEventFromJson(json);

}
```

error

```
  LibraryGenerationResultKind.noChange
    The null object does not have a getter 'parameters'.

    NoSuchMethodError: method not found: 'parameters'
    Receiver: null
    Arguments: []
    #0      Object._noSuchMethod (dart:core-patch/object_patch.dart:42)
    #1      Object.noSuchMethod (dart:core-patch/object_patch.dart:45)
    #2      _writeFactory (package:source_gen/generators/json_serializable_generator.dart:104:24)
    #3      JsonSerializableGenerator.generateForAnnotatedElement.<generateForAnnotatedElement_async_body> (package:source_gen/generators/json_serializable_generator.dart:50:20)
    #4      Future.Future.microtask.<anonymous closure> (dart:async/future.dart:144)
    #5      _microtaskLoop (dart:async/schedule_microtask.dart:41)
    #6      _startMicrotaskLoop (dart:async/schedule_microtask.dart:50)
    #7      _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:96)
    #8      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:149)

```

_Copied from original issue: dart-lang/source_gen#79_

**Comment by Aetet** [comment]
@kevmoo is any progress on this?

**Comment by kevmoo** [comment]
I'll look now.

@Aetet – what do you **want** here? A better error? Support other named ctors?

**Comment by Aetet** [comment]
Hi, @kevmoo . Better error while generation. Today it's not verbose.

**Comment by kevmoo** [comment]
@Aetet see https://github.com/dart-lang/json_serializable/pull/45

**Comment by kevmoo** [comment]
@Aetet – now published as 0.2.4+1

**Comment by Aetet** [comment]
Thanks :+1:
